source/common/http/conn_manager_impl.h

```diff
@@ -293,8 +328,14 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     void addStreamDecoderFilterWorker(StreamDecoderFilterSharedPtr filter, bool dual_filter);
     void addStreamEncoderFilterWorker(StreamEncoderFilterSharedPtr filter, bool dual_filter);
     void chargeStats(const HeaderMap& headers);
+    // Returns the encoder filter to start iteration with. If the function is called from a filter
+    // that should always iterate from the next filter, always_start_next should be set to true.
     std::list<ActiveStreamEncoderFilterPtr>::iterator
-    commonEncodePrefix(ActiveStreamEncoderFilter* filter, bool end_stream);
+    commonEncodePrefix(ActiveStreamEncoderFilter* filter, bool end_stream, bool always_start_next);
```

Comment:
Can you replace `always_start_next` with a 2 state enum and use that here and below? I think it will make the calling code significantly more self describing.

Reply:
Fixed. Thanks!